crates/xtask/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+use std::path::PathBuf;
+use structopt::StructOpt;
+use color_eyre::{Result, eyre::eyre};
+use duct::cmd;
+
+mod build;
+mod docker;
+mod install;
+
+use crate::build::BuildOpt;
+use crate::install::InstallTargetOpt;
+use crate::install::install_target;
+use crate::docker::DockerOpt;
+use serde_json::Value;
+
+pub const CARGO: &str = env!("CARGO");
+
+#[derive(StructOpt, Debug)]
+#[structopt(settings = &[structopt::clap::AppSettings::DeriveDisplayOrder])]
+pub struct Root {
+    #[structopt(subcommand)]
+    pub cmd: RootCmd,
+}
+
+#[derive(StructOpt, Debug)]
+pub enum RootCmd {
```

Comment:
This defeats purpose of xtask.  It's better if this is separated out

Reply:
Can you expand on this? To me the purpose of xtask is to move all logic associated with our build into Rust, where we can write clean and reusable code to maintain our build process better, and I don't see what `RootCmd` itself has to do with that.

Reply:
Seems like this is overloaded with many task which don't think xtask is designed for.  Ideally, there should many tasks such that they can be composed not a single monolithic super task:

For example, I can see task break down in terms of:
* Building components (CLI, platform, smart stream)
* Building docker image
* Running integration test
* Creating K8 cluster, cleaning
* Publishing
* Releasing
* etc